### PR TITLE
Archive gh actions artifacts

### PR DESCRIPTION
**Pull request by shtirlic**

### CODE REVIEW

**Comment by vitaliy-aksyonov** [comment]
Is there any retention policy? How long builds will be stored? How much space we may use?

**Comment by shtirlic** [comment]
@vitaliy-aksyonov 
```
Artifacts are retained for 90 days by default.
```

File: .github/workflows/cmake-multi-platform.yml [hunk 1]
```diff
@@ -71,3 +71,33 @@ jobs:
...
+    - uses: benjlevesque/short-sha@v2.2
+      id: short-sha
+      with:
+        length: 6
+
+    - if: matrix.os == 'windows-latest'
+      name: Zip binaries
+      run: >
+        git archive --format=zip -o golded-${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.cpp_compiler }}-${{ steps.short-sha.outputs.sha }}.zip
+        --add-file=${{ steps.strings.outputs.build-output-dir }}/golded3/golded.exe
```
Comment (vitaliy-aksyonov) [inline .github/workflows/cmake-multi-platform.yml:84]: You may still use one step for windows and others if just use mask like golded* or golded.*

**Review by vitaliy-aksyonov — CHANGES_REQUESTED** [review]

File: .github/workflows/cmake-multi-platform.yml (hunk 1 above)
Comment (shtirlic) [reply .github/workflows/cmake-multi-platform.yml:84]: Unfortunately it does not work, I tested  it locally (

Comment (shtirlic) [reply .github/workflows/cmake-multi-platform.yml:84]: and the windows paths are different

Comment (spanevin) [reply .github/workflows/cmake-multi-platform.yml]: > You may still use one step for windows and others if just use mask like golded* or golded.*

I'm not sure golded.* will work correctly, . is just a part of a filename :)
But why golded? Isn't it named gedlnx on linux?

File: .github/workflows/cmake-multi-platform.yml [hunk 2]
```diff
@@ -71,3 +72,33 @@ jobs:
...
+        --add-file=${{ steps.strings.outputs.build-output-dir }}/golded3/Release/golded.exe
+        --add-file=${{ steps.strings.outputs.build-output-dir }}/goldnode/Release/goldnode.exe
+        --add-file=${{ steps.strings.outputs.build-output-dir }}/rddt/Release/rddt.exe
+        HEAD README.md
+
+    - if: matrix.os != 'windows-latest'
+      name: Zip binaries
+      run: >
+        git archive --format=zip -o golded-${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.c_compiler }}-${{ steps.short-sha.outputs.sha }}.zip
+        --add-file=${{ steps.strings.outputs.build-output-dir }}/golded3/golded
```
Comment (spanevin) [inline .github/workflows/cmake-multi-platform.yml:94]: isn't it gedlnx?

Comment (spanevin) [reply .github/workflows/cmake-multi-platform.yml:94]: I have gedlnx, gnlnx and rddtlnx

Comment (vitaliy-aksyonov) [reply .github/workflows/cmake-multi-platform.yml:94]: It's not, when build using cmake.

Comment (spanevin) [reply .github/workflows/cmake-multi-platform.yml:94]: Why don't they produce the same artifacts? :(

Comment (shtirlic) [reply .github/workflows/cmake-multi-platform.yml:94]: legacy make did this, cmake is making straightforward names

Comment (shtirlic) [reply .github/workflows/cmake-multi-platform.yml:94]: all done https://github.com/golded-plus/golded-plus/actions/runs/6630863513
in the summary there is archive of all the artifacts from the builds

Comment (spanevin) [reply .github/workflows/cmake-multi-platform.yml:94]: I'm pretty sure there were reasons to give these names :)

**Comment by shtirlic** [comment]
Or I can make them separated

**Comment by shtirlic** [comment]
image
looks better separated

**Review by vitaliy-aksyonov — APPROVED** [review]

**Comment by dukelsky** [comment]
How do you want to use the stored Golded binaries?

**Comment by shtirlic** [comment]
Common practice is to use them for testing fixes and new features for people waiting them without the need of recompile on the end-user side as it usually introduces platform or user specific settings affecting the end-user build.

**Pull request merged**